files/bootstrap.sh

```diff
@@ -342,3 +342,31 @@ fi
 systemctl daemon-reload
 systemctl enable kubelet
 systemctl start kubelet
+
+# gpu boost clock
+if  command -v nvidia-smi &>/dev/null ; then
+   echo "nvidia-smi found"
+
+   sudo nvidia-smi -pm 1 # set persistence mode
+   sudo nvidia-smi --auto-boost-default=0
+
+   GPUNAME=$(nvidia-smi -L | head -n1)
+   echo $GPUNAME
+
+   # set application clock to maximum
+   if [[ $GPUNAME == *"A100"* ]]; then
+      nvidia-smi -ac 1215,1410
```

Comment:
What's the source of these numbers other than the other references script?

Reply:
Those values are from running [nvidia-smi -i 0 -q -d SUPPORTED_CLOCKS | grep -F 'Memory' -A1 | head -n2], which means maximum supported memory and core clock respectively, that we want to set our application clock as. It depends on GPU type we have in our gpu instances, I ran the command for each gpu type to confirm those values.

Reply:
was there an answer to:
> Does it have any implications on non-p4 instances that people might be running the nvidia driver?
we use g3/g4 instance families, and also curious if there are implications outside of p4

Reply:
Yes, impact for G4 and G3 is also included, we're setting them to achieve better performance. You can find more details from https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/optimize_gpu.html